packages/react-components/react-dialog/src/utils/useDisableBodyScroll.ts

```diff
@@ -1,55 +1,61 @@
 import { useFluent_unstable } from '@fluentui/react-shared-contexts';
-import { useCallback } from 'react';
+import { makeResetStyles } from '@griffel/react';
+import { useCallback, useRef } from 'react';
 
-const disableScrollElementProp = '__fluentDisableScrollElement' as const;
-
-type FluentDisableScrollElement = HTMLElement & {
-  [disableScrollElementProp]: {
-    count: number;
-    previousOverflowStyle: string;
-    previousPaddingRightStyle: string;
-  };
-};
+// this style must be applied to the html element to disable scrolling
+const useHTMLNoScrollStyles = makeResetStyles({
+  overflowY: ['hidden', 'clip'],
+});
 
 /**
- * hook that disables body scrolling through `overflow: hidden` CSS property
+ * hook that disables body scrolling through `overflowY: hidden` CSS property
  */
-export function useDisableBodyScroll() {
+export function useDisableBodyScroll(): {
```

Comment:
Just as a reference for further discussion:

Here's with no `scrollbar-gutter` and no `padding-right` being applied:

https://github.com/microsoft/fluentui/assets/5483269/4924686d-5d26-4fbf-bf8a-43546ff402ff

> whole content underneath the dialog shifts due to the removal of the scrollbar

Here's `padding-right` strategy:

https://github.com/microsoft/fluentui/assets/5483269/ae343c8e-2f86-49af-b66c-4a5dc3dd756a

> seems like `padding-right` solves the problem for the content shifting underneath the dialog, but for some reason the text is still shifting?!

Here's `scrollbar-gutter: stable` strategy:

https://github.com/microsoft/fluentui/assets/5483269/33e2ba69-b322-438a-8a7a-31bcd383790c

> in this case there's no more shifting of content underneath the dialog, but `scrollbar-gutter` introduces a whitespace on the right where the scrollbar was supposed to be.